pkg/provider/azure.go

```diff
@@ -315,7 +319,27 @@ func (az *AzureProvider) Decommision(node *v1.Node) error {
 }
 
 func (az *AzureProvider) getStorageAccount(account string) (storage.Account, error) {
-	return az.clients.Accounts.GetProperties(context.Background(), az.resourceGroup, account, storage.AccountExpandBlobRestoreStatus)
+	ctx := context.Background()
+	saList, err := az.clients.Accounts.List(ctx)
+	if err != nil {
+		return storage.Account{}, err
+	}
+	// check if storage account belongs to the same resource group
+	for _, sa := range saList.Values() {
+		if *sa.Name == account {
```

Comment:
Why does this work?

Reply:
Dereferencing a pointer gives us access to the value the pointer points to

Reply:
oh no not that part, ike how does the logic itself work?

Reply:
You're listing storage accounts in the subscription then somehow getting the resource group by parsing the full path name or somnething?  Is there no reference to it elsewhere?

Reply:
Wow, that's nuts. Azure really is trash then